lib/streamlit/util.py

```diff
@@ -212,3 +212,21 @@ def extract_key_query_params(
             for item in sublist
         ]
     )
+
+
+def extract_single_element_lists(
+    dict: Dict[str, List[Any]]
+) -> Dict[str, Union[Any, List[Any]]]:
+    """
+    Extracts the single element from lists in a dictionary if the list has a length of 1.
+
+    Args:
+    - dict (dict): The dictionary containing lists.
+
+    Returns:
+    - dict: The modified dictionary with single element lists extracted.
+    """
+    for key, value in dict.items():
+        if isinstance(value, list) and len(value) == 1:
+            dict[key] = value[0]
+    return dict
```

Comment:
This way this is done isn't ideal because it also modifies the input dict. We can avoid this and also write slightly more idiomatic code with a dict comprehension:

```python
return {
    k: v[0] if isinstance(v, list) and len(v) == 1 else v
    for k, v in query_params.items()}
}
```